src/client/src/Stream.c

```diff
@@ -659,31 +662,40 @@ STATUS logStreamMetric(PKinesisVideoStream pKinesisVideoStream)
 
-    DLOGD("Kinesis Video client and stream metrics: \n\t"
-                  "Overall storage byte size: %" PRIu64 "\n\t"
-                  "Available storage byte size: %" PRIu64 "\n\t"
-                  "Allocated storage byte size: %" PRIu64 "\n\t"
-                  "Total view allocation byte size: %" PRIu64 "\n\t"
-                  "Total streams frame rate (fps): %" PRIu64 "\n\t"
-                  "Total streams transfer rate (bps): %" PRIu64 " (%" PRIu64 " Kbps)\n\t"
-                  "Current view duration (ms): %" PRIu64 "\n\t"
-                  "Overall view duration (ms): %" PRIu64 "\n\t"
-                  "Current view byte size: %" PRIu64 "\n\t"
-                  "Overall view byte size: %" PRIu64 "\n\t"
-                  "Current frame rate (fps): %f\n\t"
-                  "Current transfer rate (bps): %" PRIu64 " (%" PRIu64 " Kbps)",
-          clientMetrics.contentStoreSize,
-          clientMetrics.contentStoreAvailableSize,
-          clientMetrics.contentStoreAllocatedSize,
-          clientMetrics.totalContentViewsSize,
-          clientMetrics.totalFrameRate,
-          clientMetrics.totalTransferRate * 8, clientMetrics.totalTransferRate * 8 / 1024,
-          streamMetrics.currentViewDuration / HUNDREDS_OF_NANOS_IN_A_MILLISECOND,
-          streamMetrics.overallViewDuration / HUNDREDS_OF_NANOS_IN_A_MILLISECOND,
-          streamMetrics.currentViewSize,
-          streamMetrics.overallViewSize,
-          streamMetrics.currentFrameRate,
-          streamMetrics.currentTransferRate * 8, streamMetrics.currentTransferRate * 8 / 1024);
+    DLOGD("Kinesis Video client and stream metrics:");
+    DLOGD("\tOverall storage byte size: %" PRIu64 " ", clientMetrics.contentStoreSize);
+    DLOGD("\tAvailable storage byte size: %" PRIu64 " ", clientMetrics.contentStoreAvailableSize);
+    DLOGD("\tAllocated storage byte size: %" PRIu64 " ", clientMetrics.contentStoreAllocatedSize);
+    DLOGD("\tTotal view allocation byte size: %" PRIu64 " ", clientMetrics.totalContentViewsSize);
+    DLOGD("\tTotal streams frame rate (fps): %" PRIu64 " ", clientMetrics.totalFrameRate);
+    DLOGD("\tTotal streams transfer rate (bps): %" PRIu64 " (%" PRIu64 " Kbps)", clientMetrics.totalTransferRate * 8, clientMetrics.totalTransferRate * 8 / 1024);
+    DLOGD("\tCurrent view duration (ms): %" PRIu64 " ", streamMetrics.currentViewDuration / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+    DLOGD("\tOverall view duration (ms): %" PRIu64 " ", streamMetrics.overallViewDuration / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+    DLOGD("\tCurrent view byte size: %" PRIu64 " ", streamMetrics.currentViewSize);
+    DLOGD("\tOverall view byte size: %" PRIu64 " ", streamMetrics.overallViewSize);
+    DLOGD("\tCurrent frame rate (fps): %f ", streamMetrics.currentFrameRate);
+    DLOGD("\tCurrent transfer rate (bps): %" PRIu64 " (%" PRIu64 " Kbps)", streamMetrics.currentTransferRate * 8, streamMetrics.currentTransferRate * 8 / 1024);
+
+    // V1 information
```

Comment:
I understand we need to report the metrics we calculate, but, considering the current rate at which the information is logged (at every key frame), I feel this will be a lot of information in the logs. Should we display all the V1 metrics in the logs?

Reply:
I share the concern. Would be better if we print the metric once per minute or so If it is not easily doable from PIC, better to not print anything at this layer and leave all printing to C-Producer.

Reply:
I also share this concern. Especially, since this is in PIC and especially since the produces could have multiple streams. Especially, since those streams could be audio when every frame is a key-frame and some are being produced at hundreds of FPS.

We do printout in the CPP layer too.

I would like to address this properly. Let me add some changes for handling this. I think the best would be to log on a key-frame after certain time elapses controlled by a pass-in variable. Currently, it's gated on a boolean pKinesisVideoClient->deviceInfo.clientInfo.logMetric but I will make it a period